cmd/build.go

```diff
@@ -312,23 +354,38 @@ func (c buildConfig) Prompt() (buildConfig, error) {
 	if err != nil {
 		return c, err
 	}
-	if (f.Registry == "" && c.Registry == "" && c.Image == "") || c.Confirm {
-		fmt.Println("A registry for function images is required. For example, 'docker.io/tigerteam'.")
-		err := survey.AskOne(
-			&survey.Input{Message: "Registry for function images:", Default: c.Registry},
-			&c.Registry,
-			survey.WithValidator(NewRegistryValidator(c.Path)))
-		if err != nil {
-			return c, fn.ErrRegistryRequired
+
+	// Check if function exists first
+	if !f.Initialized() {
+		// Return a specific error for uninitialized function
+		return c, fn.NewErrNotInitialized(f.Root)
+	}
+
+	// If function IS initialized AND registry/image is missing
+	if f.Registry == "" && c.Registry == "" && c.Image == "" {
+		// In interactive terminal, prompt for registry (user-friendly workflow)
+		if interactiveTerminal() {
```

Comment:
I see there might be a misunderstaning of what th e`.Prompt` method means:  It's specifically for gathering additional information from a user... which can only happen if it's currently an interactive terminal.

So placing that check at the beginning of the method is fine.